src/main/java/uk/ac/ic/wlgitbridge/snapshot/push/PostbackPromise.java

```diff
@@ -35,7 +35,7 @@ public int waitForPostback() throws SnapshotPostException {
         try {
             while (!received) {
                 try {
-                    if (!cond.await(60, TimeUnit.SECONDS)) {
+                    if (!cond.await(60 * 3, TimeUnit.SECONDS)) {
```

Comment:
I think it would be worth extracting this into a constant rather than having the magic number. That would also let us put the timeout in the message, which is a nice touch.